### PR TITLE
Add vsphere anit-affinity rule to terraform example

### PR DESCRIPTION
**Pull request by mihiragrawal**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** :

This PR adds the vsphere anti-affinity rule to terraform example so control planes can spread across the hosts. 

**Does this PR introduce a user-facing change?**:

```release-note
Add vsphere anti-affinity rule for the control plane to avoid a single point of failure. 
```

### CODE REVIEW

**Review by kron4eg — CHANGES_REQUESTED** [review]
Please add:
* proper kind (with slash)
* proper release-note (how is this a kind feature but how no release-note?)
* sync this change to vsphere-flatcar configuration too

File: examples/terraform/vsphere/main.tf [hunk 1]
```diff
@@ -164,3 +164,14 @@ resource "null_resource" "keepalived_config" {
...
+/*
+vSphere DRS requires a vSphere Enterprise Plus license. Toggle variable value off if you don't have it.
+An anti-affinity rule places a control_plane machines across different hosts within a cluster, and is useful for preventing single points of failure.
+*/
+
+resource "vsphere_compute_cluster_vm_anti_affinity_rule" "vm_anti_affinity_rule" {
+  count               = var.is_vsphere_enterprise_plus_license ? 1 : 0
+  name                = "vm-anti-affinity-rule"
+  compute_cluster_id  = data.vsphere_compute_cluster.cluster.id
+  virtual_machine_ids = [for k, v in vsphere_virtual_machine.control_plane : v.id]
```
[the remark this answers is not in the document]
Comment (mihiragrawal) [reply examples/terraform/vsphere/main.tf]: Fixed. Thanks for the suggestions.

**Pull request merged**